artifact-manifest-spec.md

```diff
@@ -0,0 +1,114 @@
+
+An artifact manifest provides configuration, a collection of blobs and an optional collection of references to other artifacts.
+
+- **`schemaVersion`** *int*
+
+  This REQUIRED property specifies the artifact manifest schema version.
+  For this version of the specification, this MUST be `1`. The value of this field WILL change as the manifest schema evolves. Minor version changes to the `oci.artifact.manifest` spec MUST be additive, while major version changes MAY be breaking. Artifact clients MUST implement version checking to allow for future, yet unknown changes. Artifact clients MUST ignore additive properties to minor versions. Artifact clients MAY support major changes, with no guarantee major changes MAY impose breaking changing behaviors. Artifact authors MAY support new and older schemaVersions to provide the best user experience.
+
+- **`mediaType`** *string*
+
+  This field contains the `mediaType` of this document, differentiating from [image-manifest][oci-image-manifest-spec] and [oci-image-index]. The mediaType for this manifest type MUST be `application/vnd.oci.artifact.manifest.v1+json`, where the version WILL change to reflect newer versions. Artifact authors SHOULD support multiple `mediaType` versions to provide the best user experience for their artifact type.
+
+- **`artifactType`** *string*
+
+  A unique value, [defining the type of unique artifact type][artifact-type], as registered with iana.org. See [registering unique types.][registering-iana]. OCI Artifacts that used `oci.image.manifest` used the `manifest.config.mediaType` to differentiate the type of artifact, differentiating from a container image. Artifact authors that implement `oci.artifact.manifest` use `artifactType` to differentiate the type of artifact, freeing up the `config.mediaType` to be OPTIONAL, or shared config schemas between multiple artifact types.
+
+- **`config`** *[descriptor](descriptor.md)*
+
+    This OPTIONAL property references the configuration object for an artifact. When provided, the configuration object MUST be defined with a [descriptor][descriptor], but MAY be any schema defined by the artifact author. As with any descriptor, the `descriptor.mediaType` MUST be registered with iana.org for uniqueness. The `config.mediaType` MAY use any existing schema, and artifact authors are encouraged to utilize existing schemas to simplify artifact implementations.
+
+- **`blobs`** *array of objects*
+
+    A collection of blobs, that represent a collection of 1 or more files. The blobs array is analogous to [oci.image.manifest layers][oci-image-manifest-spec-layers], however unlike [image-manifest][oci-image-manifest-spec], the ordering of blobs is specific to the artifact type. Some artifacts may choose an overlay of files, while other artifact types may store collections of files in different locations. To reflect the artifact specific implementations, the `oci.image.manifest.layers` collection is renamed to `blobs`.
```

Comment:
Can we allow the array to be empty, supporting artifact types that do not require any blobs. At present it says "1 or more" so I'm looking for "0 or more" or to mark this as "optional".

Reply:
Yes, this is a great point. Some artifacts may have 1 or more blobs, with no config. Other artifacts may have a config, no blobs, while other artifacts may have meta-data that links multiple artifacts. 
Having that flexibility in the `artifact.manifest` enables artifact authors to define the rules for their artifact type.

Reply:
See latest `artifact-manifest-spec.md` as I've removed config, and provide context for how it can be one of the many `[blobs]`, using the `mediaType` to identify it's a config, vs. content blob.